### PR TITLE
add basic button

### PR DESCRIPTION
**Pull request by simowaer**
# Button

- Exports a button as a package with specific props but without any styling, this is specific to each project
- All styling lives inside each project. We override defaults with color variables on `:root`
- Left default styling and `ant` prefixes in for now (just to show how it works).
- For the demo I suggest we use this catalog layout: https://www.interactivethings.com/styleguide/#/components/button

## Why is this a separate package? I can just define `<button>` inside my project?

- We need a standard implementation to build upon
- Built-in types
- Built-in tests (skipped for now, but available in `ant`)
- Built-in loading styling
- Built-in fallback on `href` prop to `<a>` element
- Built-in spacing between icons, text and other children of button
- Built-in A11Y (WAI-ARIA roles and attributes)
- Built-in `ButtonGroup`

## Todo

- [x] Add README to itp-rcc-button
- [x] Publish `@inthepocket/itp-rcc-button` to npm

image

### CODE REVIEW

File: packages/design-docs/package.json [hunk 1]
```diff
@@ -29,6 +29,7 @@
     "stylelint-config-standard": "^18.2.0"
   },
   "dependencies": {
+    "@inthepocket/itp-rcc-button": "^0.0.1",
```
Comment (easybird) [inline packages/design-docs/package.json]: could you publish to npm before merging? currently you are using npm link to test I suppose

File: packages/itp-rcc-button/src/Button.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+}
+
+export default class Button extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      loading: props.loading,
+    }
+
+    this.handleClick.bind(this)
```
Comment (easybird) [inline packages/itp-rcc-button/src/Button.js]: I rather prefer not to bind this, but to use an arrow function for the handleClick method

File: packages/itp-rcc-button/src/Button.js [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.delayTimeout) {
+      clearTimeout(this.delayTimeout)
+    }
+  }
+
+  handleClick(e) {
```
Comment (easybird) [inline packages/itp-rcc-button/src/Button.js]: arrow function instead of binding in constructor

File: packages/itp-rcc-button/src/Button.js (hunk 2 above)
Comment (easybird) [reply packages/itp-rcc-button/src/Button.js]: In that case you can get rid of the constructor and use ```state = { loading: props.loading }```

File: packages/itp-rcc-button/src/Button.js [hunk 4]
```diff
@@ -0,0 +1,178 @@
+import React, {Component} from 'react'
```
Comment (easybird) [inline packages/itp-rcc-button/src/Button.js]: is this Button the copy from the Typescript Ant Design one? Or homemade by you? If it's a copy, please mention this + link on top of the component

Comment (easybird) [reply packages/itp-rcc-button/src/Button.js]: - [ ] if copy, please mention

Comment (simowaer) [reply packages/itp-rcc-button/src/Button.js]: I think we should indeed mention if only to maintain the base button. Had to make some changes to make it work though.

File: packages/itp-rcc-button/src/Button.js (hunk 2 above)
Comment (simowaer) [reply packages/itp-rcc-button/src/Button.js]: Agreed, same comment as below. We'll need to check changes needed in `itp-react-scripts`

File: packages/itp-rcc-button/src/Button.js (hunk 3 above)
Comment (easybird) [reply packages/itp-rcc-button/src/Button.js]: this is strange, as in daikin we use it all over without overwriting the defaults.. could you for now overwrite those rules in the eslintrc in the project? Then idd, we can check to change the defaults inside itp-react-script if necessary

Comment (simowaer) [reply packages/itp-rcc-button/src/Button.js]: client project was at `^0.44.2`, comes with `babel-eslint`, updating `package.json`...

**Pull request merged**